logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.logserver;
+
+import alluxio.util.CommonUtils;
+import alluxio.util.WaitForOptions;
+
+import org.apache.log4j.Hierarchy;
+import org.apache.log4j.Level;
+import org.apache.log4j.PropertyConfigurator;
+import org.apache.log4j.spi.LoggerRepository;
+import org.apache.log4j.spi.RootLogger;
+
+import com.google.common.base.Function;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.NumberFormatException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Properties;
+
+/**
+ * A centralized log server for Alluxio
+ *
+ * Alluxio masters and workers generate logs and store the logs in local storage. {@link AlluxioLogServerProcess}
+ * allows masters and workers to "push" their logs to a centralized log server where another copy of the logs will
+ * be stored.
+ */
+public class AlluxioLogServerProcess implements LogServerProcess {
+  private ServerSocket mServerSocket;
+  private int mPort;
+  private String mBaseLogDir;
+  private HashMap<InetAddress, Hierarchy> mInetAddressHashMap;
```

Comment:
Done.

Reply:
I have removed this map in the current version. The purpose of this Map is to optimize for many but short-living client connections so that each time the client with the same IP connects, log server does not have to configure the logger for this client. This case does not exist in our environment since client connections are long-lived.